### PR TITLE
Added implementation for MSD radix sort algorithm based on binary representation

### PR DESCRIPTION
**Pull request by Txbias**
### **Describe your change:**
Added implementation for the MSD radix  sort algorithm. The sorting is based on the binary 
representation of the numbers.  

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: sorts/msd_radix_sort.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    1 at the end.
+    :param list_of_ints: A list of integers
+    :param bit_position: the position of the bit that gets compared
+    :return: Returns a partially sorted list
+    >>> _msd_radix_sort([45, 2, 32], 1)
+    [2, 32, 45]
+    >>> _msd_radix_sort([10, 4, 12], 2)
+    [4, 12, 10]
+    """
+    if bit_position == 0 or len(list_of_ints) == 1 or len(list_of_ints) == 0:
```
Comment (mateuszz0000) [inline sorts/msd_radix_sort.py:48]: Calling `len()` two times here is inefficient. Let's imagine 1kk list :)

`len(list_of_ints) in (0, 1)`

File: sorts/msd_radix_sort.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        return list_of_ints
+
+    zeros = list()
+    ones = list()
+    # Split numbers based on bit at bit_position from the right
+    for index, number in enumerate(list_of_ints):
```
Comment (mateuszz0000) [inline sorts/msd_radix_sort.py:54]: index never used so go without `enumerate`

File: sorts/msd_radix_sort.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    >>> msd_radix_sort([123, 345, 123, 80])
+    [80, 123, 123, 345]
+    >>> msd_radix_sort([1209, 834598, 1, 540402, 45])
+    [1, 45, 1209, 540402, 834598]
+    """
+    if not list_of_ints:
+        return []
+
+    if min(list_of_ints) < 0:
+        raise ValueError("All numbers must be positive")
```
Comment (mateuszz0000) [inline sorts/msd_radix_sort.py:29]: Please add test for this assertion.

**Review by mateuszz0000 — CHANGES_REQUESTED** [review]

**Comment by Txbias** [comment]
Hey, 
thank you for your feedback.
I fixed the issues you mentioned.

**Review by mateuszz0000 — APPROVED** [review]
Nice

**Pull request merged**